File: elovate_backend/src/main/java/com/tuwien/elovate/dtos/rating/Glicko2RatingParametersDto.java
package com.tuwien.elovate.dtos.rating;

/*-
 * #%L
 * ELOvate
 * %%
 * Copyright (C) 2024 ELOvate GmbH.
 * %%
 * Copyright (C) 2024 ELOvate GmbH. - All Rights Reserved
 * Unauthorized copying of this file, via any medium is strictly prohibited
 * Proprietary and confidential
 * #L%
 */

import lombok.Data;

@Data
public class Glicko2RatingParametersDto {

    Long id;
    Double defaultRating;
    Double defaultRatingDeviation;
    Double defaultRatingVolatility;
    Double tau;
    Integer ratingPeriod; // this is in days - the mapper maps this to the valid Period Object
    Double minRating;
    Double maxRating;

}

File: elovate_frontend/src/dtos/rating/glicko2RatingParametersDto.ts
/*-
 * #%L
 * ELOvate
 * %%
 * Copyright (C) 2024 ELOvate GmbH.
 * %%
 * Copyright (C) 2024 ELOvate GmbH. - All Rights Reserved
 * Unauthorized copying of this file, via any medium is strictly prohibited
 * Proprietary and confidential
 * #L%
 */
export interface Glicko2RatingParametersDto {
  defaultRating: number;
  defaultRatingDeviation: number;
  defaultRatingVolatility: number;
  tau: number;
  ratingPeriod: number; // in days
  minRating: number;
  maxRating: number;
}
